lib/block_view_message.go

```diff
@@ -308,52 +577,396 @@ func (bav *UtxoView) _disconnectPrivateMessage(
 	// Now we know the txMeta is PrivateMessage
 	txMeta := currentTxn.TxnMeta.(*PrivateMessageMetadata)
 
-	// Get the MessageEntry for the sender in the transaction. If we don't find
-	// it or if it has isDeleted=true that's an error.
-	senderMessageKey := MakeMessageKey(currentTxn.PublicKey, txMeta.TimestampNanos)
-	messageEntry := bav._getMessageEntryForMessageKey(&senderMessageKey)
-	if messageEntry == nil || messageEntry.isDeleted {
+	// Check for the message version in transaction's ExtraData.
+	version := uint64(1)
+	if extraV, hasExtraV := currentTxn.ExtraData["V"]; hasExtraV {
+		var err error
+		rr := bytes.NewReader(extraV)
+		version, err = ReadUvarint(rr)
+		if err != nil {
+			return errors.Wrapf(RuleErrorPrivateMessageInvalidVersion,
+				"_disconnectPrivateMessage: Problem reading version from ExtraData, error: (%v)", err)
+		}
+		if version > 3 {
+			return errors.Wrapf(RuleErrorPrivateMessageInvalidVersion,
+				"_disconnectPrivateMessage: Problem reading version from ExtraData")
+		}
```

Comment:
same as mentioned above. we should validate that the version is between 1 and 3. or better yet, define a function called `isMessageVersionValid` or something. 

Also this code is duplicated from above - can we refactor?

Reply:
good, cleaner code! I like